src/preset.js

```diff
@@ -8,25 +8,153 @@
+ */
+genotet.preset.NETWORK_NAME_ = 'My Network';
+/**
+ * Name for preset expression view.
+ * @private @const {string}
+ */
+genotet.preset.EXPRESSION_NAME_ = 'My Expression Matrix';
+/**
+ * Name for preset binding view.
+ * @private @const {string}
+ */
+genotet.preset.BINDING_NAME_ = 'My Genome Browser';
+/**
+ * Name for preset 3-track binding view.
+ * @private @const {string}
+ */
+genotet.preset.THREE_TRACK_BINDING_NAME_ = 'My 3 Tracks Genome Browser';
+
+/**
  * Loads a preset with the given name.
  * @param {string} preset
  */
 genotet.preset.loadPreset = function(preset) {
+  genotet.viewManager.closeAllViews();
+  var expressionType = genotet.data.VIEW_TYPES[0];
+  var networkType = genotet.data.VIEW_TYPES[1];
+  var bindingType = genotet.data.VIEW_TYPES[2];
+
   if (!preset) {
     preset = 'default';
   }
-
   switch (preset) {
     case 'default':
+      genotet.preset.createView_(networkType,
+        genotet.preset.NETWORK_NAME_, genotet.preset.NETWORK_PARAMS_);
+      genotet.preset.createView_(expressionType,
+        genotet.preset.EXPRESSION_NAME_, genotet.preset.EXPRESSION_PARAMS_);
+      genotet.preset.createView_(bindingType,
+        genotet.preset.BINDING_NAME_, genotet.preset.BINDING_PARAMS_);
       break;
     case 'network':
+      genotet.preset.createView_(networkType,
+        genotet.preset.NETWORK_NAME_, genotet.preset.NETWORK_PARAMS_);
+      genotet.preset.createView_(bindingType,
+        genotet.preset.BINDING_NAME_, genotet.preset.BINDING_PARAMS_);
       break;
     case 'expression':
+      genotet.preset.createView_(networkType,
+        genotet.preset.NETWORK_NAME_, genotet.preset.NETWORK_PARAMS_);
+      genotet.preset.createView_(expressionType,
+        genotet.preset.EXPRESSION_NAME_, genotet.preset.EXPRESSION_PARAMS_);
       break;
     case 'binding':
+      genotet.preset.createView_(networkType,
+        genotet.preset.NETWORK_NAME_, genotet.preset.NETWORK_PARAMS_);
+      genotet.preset.createView_(expressionType,
+        genotet.preset.EXPRESSION_NAME_, genotet.preset.EXPRESSION_PARAMS_);
+      genotet.preset.createView_(bindingType,
+        genotet.preset.THREE_TRACK_BINDING_NAME_,
+        genotet.preset.THREE_TRACK_BINDING_PARAMS_);
       break;
     default:
       genotet.error('unknown preset:', preset);
       return;
   }
+  genotet.linkManager.link();
+};
+
+/**
+ * Creates a preset view.
+ * @param {string} type Type for preset view.
+ * @param {string} viewName Name for preset view.
+ * @param {!genotet.ExpressionViewParams|!genotet.NetworkViewParams|
+ *   !genotet.BindingViewParams} params
+ * @private
+ */
+genotet.preset.createView_ = function(type, viewName, params) {
+  genotet.viewManager.createView(type, viewName, params);
+
+  var newView = genotet.viewManager.views[viewName];
+  switch (type) {
+    case 'network':
+      genotet.linkManager.register(newView);
+      break;
+    case 'expression':
+      for (var linkViewName in genotet.linkManager.links) {
```

Comment:
You seem to be misunderstanding stuffs so you are complaining about my "changes". In fact I didn't change any suggestions.

If you think of the links as graph edges, you would naturally store them in adjacency lists. The links object I specified above is an adjacency list externalization. It does not conflict with (action, response)'s being a list of edges.

Let me give a more detailed example:
We have two links: clicking a network node in networkA loads a binding track in bindingA, clicking a network edge in networkA loads a binding track in bindingB and also does a locus search.
We have two tuples
link1 = (networkA, nodeClick, bindingA, setFile)
link2 = (networkA, edgeClick, bindingB, setFileAndLocus)
Your link creation can take in a list of tuples [link1, link2]. This is your link input, which is handled after the creations of networkA, bindingA and bindingB.
Then you are adding the links to the views in register(). After all registrations are done, you should have the following links. 
```js
/** 
 * @type {!Object<!Array<{
 *  target: genotet.View,
 *  action: string, 
 *  response: string // apologize for missing response here
 * }>>}
 */
genotet.linkManager.links = {
  'networkA': [
    {target: bindingA, action: 'nodeClick', response: 'setFile'},
    {target: bindingB, action: 'edgeClick', response: 'setFileAndLocus'}
  ]
}
```